bolt/src/main/java/com/slack/api/bolt/service/builtin/AmazonS3InstallationService.java

```diff
@@ -32,16 +37,24 @@ public AmazonS3InstallationService(String bucketName) {
     public Initializer initializer() {
         return (app) -> {
             // The first access to S3 tends to be slow on AWS Lambda.
-            AWSCredentials credentials = getCredentials();
-            if (credentials == null || credentials.getAWSAccessKeyId() == null) {
+            this.credentialsProvider = DefaultCredentialsProvider.create();
+            AwsCredentials credentials = createCredentials(credentialsProvider);
+            if (credentials == null || credentials.accessKeyId() == null) {
                 throw new IllegalStateException("AWS credentials not found");
             }
             if (log.isDebugEnabled()) {
-                log.debug("AWS credentials loaded (access key id: {})", credentials.getAWSAccessKeyId());
+                log.debug("AWS credentials loaded (access key id: {})", credentials.accessKeyId());
+            }
+            boolean bucketExists = false;
+            Exception error = null;
+            try (S3Client s3 = createS3Client()) {
+                bucketExists = s3.headBucket(HeadBucketRequest.builder().bucket(bucketName).build()) != null;
+            } catch (Exception e) { // NoSuchBucketException etc.
+                error = e;
             }
-            boolean bucketExists = createS3Client().doesBucketExistV2(bucketName);
             if (!bucketExists) {
-                throw new IllegalStateException("Failed to access the Amazon S3 bucket (name: " + bucketName + ")");
+                String message = "Failed to access the Amazon S3 bucket (name: " + bucketName + ", error: " + error + ")";
```

Comment:
Will this print out only the `error` message or the message with the stack trace?

Do we want to have the message of this error contain the trace of another error 🤔

Reply:
This prints only this exception, but it depends on how toString method is implemented by an exception class. So, I've tweaked the code to have only the class name and message. Thanks for the suggestion!